### PR TITLE
CB-2867 support dataset in UI

### PR DESCRIPTION
**Pull request by alexander-skoblikov**

### CODE REVIEW

File: webapp/packages/plugin-data-viewer/src/TableViewer/TableViewer.tsx [hunk 1]
```diff
@@ -192,6 +192,19 @@ export const TableViewer = observer<Props>(function TableViewer({
     onValuePresentationChange,
   }, ['setPresentation', 'setValuePresentation', 'switchValuePresentation', 'closeValuePresentation']);
 
+  const needRefresh = getComputed(() => (
```
Comment (devnaumov) [inline webapp/packages/plugin-data-viewer/src/TableViewer/TableViewer.tsx:195]: Please cast dataModel to a boolean !!dataModel  so needRefresh will return boolean, not boolean | undefined

File: webapp/packages/plugin-resource-manager/src/Tree/ProjectsRenderer/navigationTreeResourceTypeFilter.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  resourceTypeId?: string,
+): IElementsTreeFilter {
+  return (filter: string, node: NavNode, children: string[]) => {
+    if (resourceTypeId === undefined) {
+      return children;
+    }
+
+    const nodes = navNodeInfoResource
+      .get(resourceKeyList(children))
+      .filter<NavNode>((node => node !== undefined) as (node: NavNode | undefined) => node is NavNode)
```
Comment (devnaumov) [inline webapp/packages/plugin-resource-manager/src/Tree/ProjectsRenderer/navigationTreeResourceTypeFilter.ts]: Maybe use utility function filterUndefined?

File: webapp/packages/plugin-sql-editor-navigation-tab-script/src/PluginBootstrap.ts [hunk 3]
```diff
@@ -118,7 +117,12 @@ export class PluginBootstrap extends Bootstrap {
                 throw new Error('Project not found');
               }
 
-              const scriptName = `${result.name.trim()}.${SCRIPT_EXTENSION}`;
+              const resourceType = this.projectInfoResource.getResourceType(project, SCRIPTS_TYPE_ID);
+              if (!resourceType) {
+                throw new Error('Resource type not found');
```
Comment (devnaumov) [inline webapp/packages/plugin-sql-editor-navigation-tab-script/src/PluginBootstrap.ts]: Please include resource type in error message

File: webapp/packages/plugin-sql-editor/src/SqlEditor/SqlEditor.tsx [hunk 4]
```diff
@@ -119,6 +125,20 @@ export const SqlEditor = observer<ISqlEditorProps>(function SqlEditor({ state, c
 
   const disabled = getComputed(() => data.isLineScriptEmpty || data.isDisabled);
   const isActiveSegmentMode = getComputed(() => data.activeSegmentMode.activeSegmentMode);
+  const displayedEditors = getComputed(() => sqlEditorModeService.tabsContainer.getDisplayed({ state, data }).length);
+
+  useEffect(() => {
```
Comment (devnaumov) [inline webapp/packages/plugin-sql-editor/src/SqlEditor/SqlEditor.tsx:130]: Maybe better to move splitting logic to the separate hook

File: webapp/packages/plugin-sql-editor/src/SqlEditor/useSqlEditor.ts [hunk 5]
```diff
@@ -261,6 +263,21 @@ export function useSqlEditor(state: ISqlEditorTabState): ISQLEditorData {
       );
     },
 
+    async loadDatabaseDataModels(): Promise<void> {
+      const query = this.getExecutingQuery(true);
+
+      await this.executeQueryAction(
+        query,
+        async () => {
+          if ((this.dataSource?.databaseModels.length ?? 0) > 0) {
```
Comment (devnaumov) [inline webapp/packages/plugin-sql-editor/src/SqlEditor/useSqlEditor.ts]: just !!this.dataSource?.databaseModels.length

File: webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlQueryService.ts [hunk 6]
```diff
@@ -65,6 +65,49 @@ export class SqlQueryService {
     return this.statisticsMap.get(tabId);
   }
 
+  executeEditorDatabaseDataModels(editorState: ISqlEditorTabState) {
+    const dataSource = this.sqlDataSourceService.get(editorState.editorId);
+    const databaseModels = dataSource?.databaseModels;
+
+    if (!databaseModels) {
+      console.error('databaseModels executionContext is not provided');
```
Comment (devnaumov) [inline webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlQueryService.ts]: Maybe better to throw an error as we return from function in this case

**Review by devnaumov — APPROVED** [review]

File: webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlQueryService.ts (hunk 6 above)
Comment (Wroud) [reply webapp/packages/plugin-sql-editor/src/SqlResultTabs/SqlQueryService.ts]: i will remove it

**Pull request merged**